File: Splunk Interview/1. Two Sum.js
var twoSum = function(nums, target) {

    let hashmap = {};
    for (let i = 0; i < nums.length; i++) {
        if ((target - nums[i] in hashmap)) return [hashmap[target-nums[i]], i];
        hashmap[nums[i]] = i;
    }
}

// o(n) solution here

File: Daily Coding Practice/2019-12-06/1 Two Sum.py
import collections
class Solution:
    def twoSum(self, nums, target):
        seen = dict()

        for idx, num in enumerate(nums):

            x = target - num
            if x in seen:
                return [seen[x], idx]
            seen[num] = idx

# time complexity o(n)
# space complexity o(n)